packages/pass-style/src/typeGuards.js

```diff
@@ -107,16 +107,17 @@ export const assertRemotable = (
 harden(assertRemotable);
 
 /**
- * @param {Passable} val
+ * @param {any} val Not necessarily passable
  * @param {Checker} check
- * @returns {boolean}
+ * @returns {val is Atom}
  */
 const checkAtom = (val, check) => {
-  // TODO There is not yet a checkPassable, but perhaps there should be.
-  // If that happens, we should call it here instead. As it is now,
-  // any non-passable will be rejected by `passStyleOf` with a thrown
-  // error.
-  const passStyle = passStyleOf(val);
+  let passStyle;
+  try {
+    passStyle = passStyleOf(val);
+  } catch (err) {
+    return check(false, X`Not even Passable: ${q(err)}: ${val}`);
+  }
```

Comment:
At least on V8, this will be considerably faster if we can avoid the try/catch. Probably considerably more understandable if `passStyleOf` simply did not throw, or if there were a variant that did not throw.

Reply:
Hmm, I also see opportunities for this and other functions to use `canBeValid` and other fast checks to avoid the full `passStyleOf` traversal. Let's leave this for now and open an issue.

Reply:
Yeah, that's been a longstanding TODO. Not for this PR.

Resolving.